metainsuranceorg.py

```diff
@@ -212,8 +212,27 @@ def enter_bankruptcy(self, time):
         self.cash = 0                           #Cash is 0 after bankruptcy.
         self.excess_capital = 0                 #Excess of capital is 0 after bankruptcy.
         self.profits_losses = 0                 #Profits and losses are 0 after bankruptcy.
+        if self.operational:
+            self.simulation.record_bankruptcy()
         self.operational = False
-        self.simulation.record_bankruptcy()
+
+
+    def market_exit(self, time):
```

Comment:
This duplicates a lot of code with the `enter_bankruptcy()` method. This is messy, hard to understand, and will cause inconsistency later. I suggest to instead have one method call the other so that no code is duplicated.

Perhaps like this:
  * `market_exit()`: Causes outstanding obligations to be paid immediately. Then calls liquidate.
  * `enter_bankruptcy()`: Calls liquidate.
  * `liquidate()`: Returns remaining assets to simulation object.

Problem:
  * The agent may receive payments in the future (for whatever reason). We should think about making provisions to forward those payments to somewhere else. Perhaps to a liquidator agent (which could just be the simulation object, so that these funds are returned to the simulation object).

Reply:
Done!